oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -115,34 +141,113 @@
          logger.trace("<< provisioning server '%s'", serverRequest);
 
          server = api.serverApi().create(serverRequest);
+
+         waitServerUntilAvailable.apply(server);
+
+         Map<Integer, Integer> portsRange = getPortRangesFromList(inboundPorts);
+         List<FirewallPolicy.Rule.CreatePayload> rules = new ArrayList<FirewallPolicy.Rule.CreatePayload>();
+
+         for (Map.Entry<Integer, Integer> range : portsRange.entrySet()) {
+            FirewallPolicy.Rule.CreatePayload rule = FirewallPolicy.Rule.CreatePayload.builder()
+                    .portFrom(range.getKey())
+                    .portTo(range.getValue())
+                    .protocol(Types.RuleProtocol.TCP)
+                    .build();
+            rules.add(rule);
+         }
+         FirewallPolicy rule = api.firewallPolicyApi().create(FirewallPolicy.CreateFirewallPolicy.create(server.name() + " firewall policy", "desc", rules));
+         updateServer = api.serverApi().get(server.id());
+         api.serverApi().addFirewallPolicy(updateServer.id(), updateServer.ips().get(0).id(), rule.id());
+         waitServerUntilAvailable.apply(server);
+
          logger.trace(">> provisioning complete for server. returned id='%s'", server.id());
 
       } catch (Exception ex) {
          logger.error(ex, ">> failed to provision server. rollbacking..");
          throw Throwables.propagate(ex);
       }
 
-      return new NodeAndInitialCredentials<Server>(server, server.id(), null);
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+              .user(loginUser)
+              .password(password)
+              .privateKey(privateKey)
+              .build();
```

Comment:
Then let's keep them for the moment. We'll cleanup the node access configuration in another pull request.